src/workers/parser/getOutOfScopeLocations.js

```diff
@@ -38,26 +40,63 @@ function getLocation(func) {
 }
 
 /**
- * Reduces an array of locations to remove items that are completely enclosed
- * by another location in the array.
+ * Find the nearest location containing the input position and
+ * return new locations without inner locations under that nearest location
+ *
+ * @param locations Notice! The locations MUST be sorted by `sortByStart`
+ *                  so that we can do linear time complexity operation.
  */
-function removeOverlaps(
-  locations: AstLocation | AstLocation[],
-  location: AstLocation
-) {
-  // support reducing without an initializing array
-  if (!Array.isArray(locations)) {
-    locations = [locations];
+function removeInnerLocations(locations: AstLocation[], position: AstPosition) {
+  // First, let's find the nearest position-enclosing function location,
+  // which is to loop locations to find the last location enclosing the bp.
+  let nearestPos = findLastIndex(locations, loc =>
```

Comment:
this naming is a bit confusing, how about `parentIndex` or `outerStartIndex` (because nearest can also be adjacent, and Pos suggests Position similar to the ASTPosition, but this is an index. more specifically it is a boundary)

also we use it twice with +1; it makes sense to have that as a variable, maybe

```javascript
const outerStartIndex = findLastIndex(locations, loc => containsPosition(loc, position));
const innerStartIndex = outerStartIndex + 1;
```